infra/repo_manager.py

```diff
@@ -146,6 +146,35 @@ def get_commit_list(self, old_commit, new_commit):
     commits.append(old_commit)
     return commits
 
+  def _get_git_history(self):
+    """Gets the current git repository history."""
+    git_path = os.path.join(self.repo_dir, '.git', 'shallow')
+    if os.path.exists(git_path):
+      build_specified_commit.execute(['git', 'fetch', '--unshallow'],
+                                     self.repo_dir,
+                                     check_result=True)
+
+  def checkout_pr(self, pr_ref):
+    """Checks out a remote pull request.
+
+    Args:
+      pr_ref: The pull request reference to be checked out.
+
+    Raises:
+      RepoManagerError: when pull request checkout fails.
+    """
+    self._get_git_history()
+
+    _, return_code = build_specified_commit.execute(
+        ['git', 'fetch', 'origin', pr_ref], self.repo_dir)
```

Comment:
Instead of checking the result and raising an error, why not specify `check_result=True` and let the exceptions happen automatically?

Reply:
Fixed.